bin/interactive.py

```diff
@@ -151,29 +148,20 @@ def run_interactive_testcase(
     # - Close remaining program + write end of pipe
     # - Close remaining read end of pipes
 
-    def mkpipe():
-        # TODO: is os.O_CLOEXEC needed here?
-        r, w = os.pipe2(os.O_CLOEXEC)
-        F_SETPIPE_SZ = 1031
-        fcntl.fcntl(w, F_SETPIPE_SZ, BUFFER_SIZE)
-        return r, w
+    def set_pipe_size(pipe):
+        # Somehow, setting the pipe buffer size is necessary to avoid hanging in larger interactions.
+        # Note that this is equivalent to Popen's `pipesize` argument in Python 3.10,
+        # but we backported this for compatibility with older Python versions:
+        # https://github.com/python/cpython/pull/21921/files#diff-619941af4b328b6abf2dc02c54e774fc17acc1ac4172c14db27d6097cbbff92aR1590
+        # See also: https://github.com/RagnarGrootKoerkamp/BAPCtools/pull/251#issuecomment-1609353538
+        fcntl.fcntl(pipe, fcntl.F_SETPIPE_SZ, BUFFER_SIZE)
```

Comment:
Can you also preserve the `CLOEXEC`? Or actually, maybe `subprocess.Popen` already handles that automatically for us.

Reply:
Yes. Probably we're good because of `subprocess` somehow.